coderd/apikey.go

```diff
@@ -516,3 +604,65 @@ func (api *API) createAPIKey(ctx context.Context, params apikey.CreateParams) (*
 		HttpOnly: true,
 	}), &newkey, nil
 }
+
+func normalizeTokenScopes(scope codersdk.APIKeyScope, scopes []codersdk.APIKeyScope) (database.APIKeyScopes, error) {
+	// Default to coder:all for backward compatibility when nothing is provided.
+	if len(scopes) == 0 && string(scope) == "" {
+		return database.APIKeyScopes{database.ApiKeyScopeCoderAll}, nil
+	}
```

Comment:
We should error if the user tries to set both right?

```suggestion
	// Default to coder:all for backward compatibility when nothing is provided.
	if len(scopes) == 0 && string(scope) == "" {
		return database.APIKeyScopes{database.ApiKeyScopeCoderAll}, nil
	}
	
	if len(scopes) > 0 && string(scope) != "" {
		return nil, xerrors.New("provide either scope or scopes, not both")
	}
```